examples/witness_verification/src/main.rs

```diff
@@ -10,8 +10,36 @@ use nomt_core::{
 pub struct Blake3Hasher;
 
 impl NodeHasher for Blake3Hasher {
-    fn hash_node(data: &nomt_core::trie::NodePreimage) -> [u8; 32] {
-        blake3::hash(data).into()
+    fn hash_leaf(data: &trie::LeafData) -> [u8; 32] {
+        let mut hasher = blake3::Hasher::new();
+        hasher.update(&data.key_path);
+        hasher.update(&data.value_hash);
+        let mut hash: [u8; 32] = hasher.finalize().into();
+
+        // Label with MSB
+        hash[0] |= 0b10000000;
+        hash
+    }
+
+    fn hash_internal(data: &trie::InternalData) -> [u8; 32] {
+        let mut hasher = blake3::Hasher::new();
+        hasher.update(&data.left);
+        hasher.update(&data.right);
+        let mut hash: [u8; 32] = hasher.finalize().into();
+
+        // Label with MSB
+        hash[0] &= 0b01111111;
+        hash
+    }
+
+    fn node_kind(node: &Node) -> NodeKind {
+        if node[0] >> 7 == 1 {
+            NodeKind::Leaf
+        } else if node == &TERMINATOR {
+            NodeKind::Terminator
+        } else {
+            NodeKind::Internal
+        }
```

Comment:
As this is an example it will be good to mention why we are doing all of that stuff.

1. Why are implementing a node hasher here ourselves? Just to demonstrate. Although arguably we might need to use a separate example for that.
2. Why is the code doing what it does? We should probably mention what's the deal with that but direct the user for more details to the rustdoc?

Reply:
I will follow up with a bit more refactoring. But actually we do it here because `Blake3Hasher` isn't exposed from `nomt-core`.